app/src/main/java/com/infomaniak/swisstransfer/ui/screen/onboarding/components/AnimatedOnboardingButton.kt

```diff
@@ -0,0 +1,110 @@
+/*
+ * Infomaniak SwissTransfer - Android
+ * Copyright (C) 2024 Infomaniak Network SA
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+package com.infomaniak.swisstransfer.ui.screen.onboarding.components
+
+import androidx.compose.animation.core.animateDpAsState
+import androidx.compose.animation.core.animateFloatAsState
+import androidx.compose.animation.core.tween
+import androidx.compose.foundation.layout.*
+import androidx.compose.material3.Icon
+import androidx.compose.material3.Text
+import androidx.compose.runtime.*
+import androidx.compose.ui.Alignment
+import androidx.compose.ui.Modifier
+import androidx.compose.ui.draw.alpha
+import androidx.compose.ui.res.stringResource
+import androidx.compose.ui.unit.dp
+import com.infomaniak.swisstransfer.R
+import com.infomaniak.swisstransfer.ui.components.SwissTransferButton
+import com.infomaniak.swisstransfer.ui.images.AppImages
+import com.infomaniak.swisstransfer.ui.images.icons.ArrowRight
+import com.infomaniak.swisstransfer.ui.theme.Dimens
+import com.infomaniak.swisstransfer.ui.theme.Margin
+import com.infomaniak.swisstransfer.ui.theme.SwissTransferTheme
+import com.infomaniak.swisstransfer.ui.utils.PreviewLightAndDark
+
+private const val BUTTON_ANIM_DURATION = 300
+private const val BUTTON_ANIM_DELAY = BUTTON_ANIM_DURATION / 2
+private const val BUTTON_ANIM_NO_DELAY = 0
+private val FAB_SIZE = 64.dp
+
+@Composable
+fun AnimatedOnboardingButton(isExpanded: () -> Boolean, onClick: () -> Unit) {
+    val buttonWidth by animateDpAsState(
+        targetValue = if (isExpanded()) Dimens.SingleButtonMaxWidth else FAB_SIZE,
+        animationSpec = tween(durationMillis = BUTTON_ANIM_DELAY),
+        label = "Onboarding button width",
+    )
+
+    val buttonHeight by animateDpAsState(
+        targetValue = if (isExpanded()) Dimens.LargeButtonHeight else FAB_SIZE,
+        animationSpec = tween(durationMillis = BUTTON_ANIM_DELAY),
+        label = "Onboarding button height",
+    )
+
+    val arrowDelay = if (isExpanded()) BUTTON_ANIM_NO_DELAY else BUTTON_ANIM_DELAY
+    val arrowVisibility by animateFloatAsState(
+        targetValue = if (isExpanded()) 0f else 1f,
```

Comment:
we could use the Boolean.toFloat() here and for textVisibility

Reply:
Same as the other .toFloat(), I feel like we loose part of the meaning, even more here